runtime/common/src/evm/precompile.rs

```diff
@@ -26,41 +51,57 @@ impl<R> CentrifugePrecompiles<R> {
 	}
 }
 
-// This value is chosen to be identical to what Moonbeam, for best
-// interoperability. See
-// https://docs.moonbeam.network/builders/pallets-precompiles/precompiles/overview/#precompiled-contract-addresses
-// for details on how Moonbeam organizes precompile addresses. We will
-// follow the same namespacing.
-const DISPATCH_ADDR: H160 = addr(1025);
-
 impl<R> PrecompileSet for CentrifugePrecompiles<R>
 where
 	R: pallet_evm::Config,
 	R::RuntimeCall: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo + Decode,
 	<R::RuntimeCall as Dispatchable>::RuntimeOrigin: From<Option<R::AccountId>>,
 {
 	fn execute(&self, handle: &mut impl PrecompileHandle) -> Option<PrecompileResult> {
-		if handle.code_address() == DISPATCH_ADDR {
-			Some(pallet_evm_precompile_dispatch::Dispatch::<R>::execute(
-				handle,
-			))
-		} else {
-			None
+		match handle.code_address().0 {
+			ECRECOVER_ADDR => Some(ECRecover::execute(handle)),
+			SHA256_ADDR => Some(Sha256::execute(handle)),
+			RIPEMD160_ADDR => Some(Ripemd160::execute(handle)),
+			IDENTITY_ADDR => Some(Identity::execute(handle)),
+			MODEXP_ADDR => Some(Modexp::execute(handle)),
+			BN128ADD_ADDR => Some(Bn128Add::execute(handle)),
+			BN128MUL_ADDR => Some(Bn128Mul::execute(handle)),
+			BN128PAIRING_ADDR => Some(Bn128Pairing::execute(handle)),
+			BLAKE2F_ADDR => Some(Blake2F::execute(handle)),
+			SHA3FIPS256_ADDR => Some(Sha3FIPS256::execute(handle)),
+			DISPATCH_ADDR => Some(Dispatch::<R>::execute(handle)),
+			ECRECOVERPUBLICKEY_ADDR => Some(ECRecoverPublicKey::execute(handle)),
+			_ => None,
 		}
 	}
 
 	fn is_precompile(&self, address: H160) -> bool {
-		address == DISPATCH_ADDR
+		[
+			ECRECOVER_ADDR,
+			SHA256_ADDR,
+			RIPEMD160_ADDR,
+			IDENTITY_ADDR,
+			MODEXP_ADDR,
+			BN128ADD_ADDR,
+			BN128MUL_ADDR,
+			BN128PAIRING_ADDR,
+			BLAKE2F_ADDR,
+			SHA3FIPS256_ADDR,
+			DISPATCH_ADDR,
+			ECRECOVERPUBLICKEY_ADDR,
+		]
+		.contains(&address.0)
```

Comment:
Performance doesn't really matter with such a small sample but, as a nit, we could make this a `match` statement for improved performance and scalability:



```suggestion
		match &address.0 {
			  ECRECOVER_ADDR |
			  SHA256_ADDR |
			  RIPEMD160_ADDR |
			  IDENTITY_ADDR |
			  MODEXP_ADDR |
			  BN128ADD_ADDR |
			  BN128MUL_ADDR |
			  BN128PAIRING_ADDR |
			  BLAKE2F_ADDR |
			  SHA3FIPS256_ADDR |
			  DISPATCH_ADDR |
			  ECRECOVERPUBLICKEY_ADDR => true,
			  _ => false,
		}
```

Reply:
or just use `matches!(`